NTUEELightDance/LightDance-Editor
# EDITOR-#60 Operations for effect lists

### What is this PR for?
Complete EffectList and fix bugs for EffectList operations in GlobalSlice

### What type of PR is it?
[Bug Fix | Improvement | Feature | Documentation | Hot Fix | Refactoring]
Bug Fix | Improvement | Feature

### Todos
- [x] Add delete button for EffectList component
- [x] Connect component to GlobalSlice for operations
- [x] Improvements
    - [x] Show delete and apply button directly
    - [x] Collision warning
    - [x] Block incorrect input 

### What is the Github issue?

[EDITOR-#60 Operations for effect lists](https://github.com/NTUEELightDance/LightDance-Editor/issues/60)

### How should this be tested?

- Add effect to effectList 
- Try to apply it to current time spot and see if all frames are successfully and correctly added
- Try delete a effect by first switch the switch to delete

### Screenshots (if appropriate)
<img width="1440" alt="截圖 2022-01-30 下午8 38 13" src="https://user-images.githubusercontent.com/55495526/151699962-2df46acc-7fba-4f6d-8512-faabbf41c0e9.png">
